gift-certificate-application/service/src/test/java/com/epam/esm/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,131 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.entity.User;
+import com.epam.esm.pagination.CustomPagination;
+import com.epam.esm.repository.dao.UserDao;
+import com.epam.esm.repository.dao.impl.UserDaoImpl;
+import com.epam.esm.service.UserService;
+import com.epam.esm.service.exception.FieldValidationException;
+import com.epam.esm.service.exception.UserAlreadyExistException;
+import com.epam.esm.service.exception.UserNotFoundException;
+import com.epam.esm.service.validator.PaginationValidator;
+import com.epam.esm.service.validator.UserValidator;
+import com.epam.esm.service.validator.impl.PaginationValidatorImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class UserServiceImplTest {
+    private static final long ID = 1L;
+    private static final String FIST_NAME = "Ivan";
+    private static final String LAST_NAME = "Ivanov";
+    private static final String LOGIN = "I-Ivan";
+    private static final int PAGE = 1;
+    private static final int SIZE = 10;
+    private static final Long ENTITIES_NUMBER = 1L;
+    private UserDao userDao;
+    private User user;
+    private UserValidator userValidator;
+    private CustomPagination pagination;
+    private PaginationValidator paginationValidator;
+    private List<User> userList;
+    private UserServiceImpl userService;
+
+    @BeforeAll
+    void beforeAll() {
+        userDao = mock(UserDaoImpl.class);
+        user = new User();
+        user.setId(ID);
+        user.setFirstName(FIST_NAME);
+        user.setLastName(LAST_NAME);
+        user.setLogin(LOGIN);
+        userValidator = mock(UserValidator.class);
+        pagination = new CustomPagination();
+        pagination.setPage(PAGE);
+        pagination.setSize(SIZE);
+        paginationValidator = mock(PaginationValidatorImpl.class);
+        userList = new ArrayList<>();
+        userList.add(user);
+        userService = new UserServiceImpl(userDao, userValidator, paginationValidator);
+    }
+
+    @Test
+    void testCreate() {
+        User expectedResult = user;
```

Comment:
I suppose you can directly use user and do not create an additional variable

Reply:
It's done